pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -816,7 +823,27 @@ func translateRouteMatch(in *networking.HTTPMatchRequest) *route.RouteMatch {
 		case *networking.StringMatch_Exact:
 			out.PathSpecifier = &route.RouteMatch_Path{Path: m.Exact}
 		case *networking.StringMatch_Prefix:
-			out.PathSpecifier = &route.RouteMatch_Prefix{Prefix: m.Prefix}
+			if (model.UseIngressSemantics(vs) || model.UseGatewaySemantics(vs)) && m.Prefix != "/" {
+				if node.IstioVersion.Compare(&istioVersion114) >= 0 {
```

Comment:
Is `proxy.IstioVersion` the version of associated control-plane or istio-proxy itself? 
If it's the former, this would not work when users only upgrade the control-plane while the istio-proxy version is still <1.14

Reply:
The proxy. We have functions already for this -- IsIstioVersionGE112, etc. Would recommend using those

Reply:
Thanks but I couldn't find them, could you point out where they are?

Reply:
We have removed such version checks in Master recently. You can find an example in 1.13 https://github.com/istio/istio/blob/c2461b13e11cabb82a0d6e7aa4bcd4e015b86ad6/pilot/pkg/networking/util/util.go#L276 - Please follow similar pattern and add that method in util

Reply:
Thank you! Will add it once https://github.com/istio/proxy/pull/3795 get merged